src/_gettsim/functions/loader.py

```diff
@@ -216,31 +249,31 @@ def load_internal_aggregation_dict(variant: _AggregationVariant):
 
 def _load_aggregation_dict(
     roots: list[Path], package_root: Path, variant: _AggregationVariant
-):
+) -> dict[str, Any]:
     """
     Load a dictionary with all aggregations by group or person reachable from the given
     roots.
     """
     roots = roots if isinstance(roots, list) else [roots]
     paths = _find_python_files_recursively(roots)
 
-    # Load dictionaries
-    dicts = []
+    tree = {}
 
     for path in paths:
-        dicts.extend(_load_dicts_in_module(path, package_root, f"{variant}_"))
-
-    # Check for duplicate keys
-    all_keys = [k for dict_ in dicts for k in dict_]
-    if len(all_keys) != len(set(all_keys)):
-        duplicate_keys = list({x for x in all_keys if all_keys.count(x) > 1})
-        raise ValueError(
-            "The following column names are used more "
-            f"than once in the {variant} dictionaries: {duplicate_keys}"
+        module_name = _convert_path_to_module_name(path, package_root)
+        # TODO(@MImmesberger): Remove the removeprefix calls once the directory
+        # structure is cleaned up
+        clean_module_name = (
+            module_name.removeprefix("_gettsim__")
+            .removeprefix("taxes__")
+            .removeprefix("transfers__")
         )
+        tree_keys = get_path_from_qualified_name(clean_module_name)
+        dicts_in_module = _load_dicts_in_module(path, package_root, f"{variant}_")
```

Comment:
```suggestion
        derived_function_names = load_functions_to_derive(path, package_root, f"{variant}_")
```

Then make the check that only one dict is allowed in that function.